webapp/src/Controller/API/ProblemController.php

```diff
@@ -417,4 +422,34 @@ public function transformObject($object)
             return $problem;
         }
     }
+
+    /**
+     * GetPDFProblem
+     * @throws NonUniqueResultException
+     * @Rest\Get("/{id}/pdf")
+     * @OA\Response(
+     *     response="200",
+     *     description="Returns the given problem for this contest",
+     *     @OA\JsonContent(ref="#/components/schemas/ContestProblem")
+     * )
+     * @OA\Parameter(ref="#/components/parameters/id")
+     * @OA\Parameter(ref="#/components/parameters/strict")
+     */
+    public function getPDFProblem(int $probId): StreamedResponse
+    {
+        return $this->getBinaryFile($probId, function (
```

Comment:
This method does not exist in the API controller. Also we should be sure to use the getQueryBuilder method to get the problem, since that will check permissions.

Reply:
So I want to create a new API endpoint to get the problem's PDF text because the current API didn't have an option to look at and open the PDF file and I want to embed it on somewhere else in the system. How should I implement these changes?

Reply:
Yeah I get what you want. Do you mind if I give it a go?

Reply:
Sure, thank you for the help!